### ISSUE TITLE
Operator panicking during upgrade from Rook 1.5.8 -> 1.6.5

### ISSUE THREAD

**Issue by eknudtson** [rook/rook#8189]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

Operator crashes while applying upgrade from Rook 1.5.8 to Rook 1.6.5

**Expected behavior:**

Operator would complete the upgrade successfully.

**How to reproduce it (minimal and precise):**
Followed upgrade instructions, and upgraded operator to 1.6.5

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
```
~$ kubectl get cephcluster
NAME        DATADIRHOSTPATH   MONCOUNT   AGE      PHASE         MESSAGE                 HEALTH      EXTERNAL
rook-ceph   /var/lib/rook     3          2y171d   Progressing   Configuring Ceph Mons   HEALTH_OK   false
```
https://gist.github.com/eknudtson/8503c411a2e88ff3069029fdba2b0c1b

* Operator's logs, if necessary
https://gist.github.com/eknudtson/ce6ea984d516138b7a38dc1bb2b06cf3

* Crashing pod(s) logs, if necessary
See operator logs

**Environment**:
* OS (e.g. from /etc/os-release): Centos 7
* Kernel (e.g. `uname -a`): 3.10.0-1160.11.1.el7.x86_64
* Cloud provider or hardware configuration: Baremetal
* Rook version (use `rook version` inside of a Rook Pod): 1.5.8, attempting to upgrade to 1.6.5
* Storage backend version (e.g. for ceph do `ceph -v`): v14.2.21-20210513
* Kubernetes version (use `kubectl version`): v1.19.7
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): baremetal
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)): HEALTH_OK

**Comment by travisn** [comment]
Here's the crash of interest from the logs:
```
2021-06-23 18:53:13.885939 I | ceph-object-controller: created object store "object-store" in namespace "rook-ceph"
2021-06-23 18:53:14.135946 I | ceph-object-controller: starting rgw healthcheck
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d32433]

goroutine 3146 [running]:
github.com/rook/rook/pkg/operator/ceph/object.(*bucketChecker).checkObjectStoreHealth(0xc007490000, 0xc0015a40c0, 0xc000f55fb0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/health.go:141 +0x1173
github.com/rook/rook/pkg/operator/ceph/object.(*bucketChecker).checkObjectStore(0xc007490000, 0xc002cd22a0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/health.go:82 +0x45
created by github.com/rook/rook/pkg/operator/ceph/object.(*ReconcileCephObjectStore).startMonitoring
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/controller.go:542 +0x3e5
```

@eknudtson This is fixed by #8139 and we will plan on the v1.6.6 release tomorrow with it. If this is blocking your cluster, we can get an interim build sooner...

**Comment by eknudtson** [comment]
Our cluster is still in a healthy state, though the operator pod is crashlooping.  I'd prefer whatever you think the best route is, I can always scale our operator deployment to 0 replicas for now.

Thanks!

**Comment by travisn** [comment]
Right, the crashing operator isn't good, but at least the data plane is safe with Ceph. An interim release build just finished and pushed to `rook/ceph:v1.6.5-40.g564be28`. This contains the fix and will essentially be the same as v1.6.6 tomorrow. It's technically not a supported release, but in reality I would expect it to be just as stable as the official releases since it ran the same test suites.

**Comment by eknudtson** [comment]
1.6.6 did the trick! Thanks for the quick response on that!
